crates/db/src/commands.rs

```diff
@@ -93,16 +94,29 @@ pub async fn db_get_contract_abi(
 
 #[tauri::command]
 pub async fn db_insert_contract(
-    chain_id: u64,
+    chain_id: u32,
     address: Address,
     db: tauri::State<'_, Db>,
 ) -> Result<()> {
-    let name = fetch_etherscan_contract_name(chain_id.into(), address).await?;
-    let abi = fetch_etherscan_abi(chain_id.into(), address)
+    let network = Networks::read()
+        .await
+        .get_network(chain_id)
+        .ok_or(Error::InvalidNetwork(chain_id))?;
+
+    if network.is_dev().await {
+        db.insert_contract_with_abi(chain_id, address, None, None)
+            .await?;
+        ethui_broadcast::ui_notify(UINotify::ContractsUpdated).await;
+
+        return Ok(());
+    }
+
+    let name = fetch_etherscan_contract_name((chain_id as u64).into(), address).await?;
+    let abi = fetch_etherscan_abi((chain_id as u64).into(), address)
```

Comment:
how is this related the the change in the PR? I can't understand the purpose here